libs/output-mapping/src/Writer/Table/TableDefinition/TableDefinitionFactory.php

```diff
@@ -4,16 +4,60 @@
 
 namespace Keboola\OutputMapping\Writer\Table\TableDefinition;
 
+use Keboola\Datatype\Definition\Exasol;
+use Keboola\Datatype\Definition\Snowflake;
+use Keboola\Datatype\Definition\Synapse;
+
 class TableDefinitionFactory
 {
+    public const NATIVE_TYPE_METADATA_KEY = 'kbc.datatype.source';
+
+    public const NATIVE_BACKEND_TYPE_CLASS_MAP = [
+        'snowflake' => Snowflake::class,
+        'synapse' => Synapse::class,
+        'exasol' => Exasol::class,
+    ];
+
+    private array $tableMetadata;
+
+    private string $backendType;
+
+    public function __construct(array $tableMetadata, string $backendType)
+    {
+        $this->tableMetadata = $tableMetadata;
+        $this->backendType = $backendType;
+    }
+
     public function createTableDefinition(string $tableName, array $primaryKeys, array $columnMetadata): TableDefinition
     {
-        $tableDefinition = new TableDefinition();
+
+        $tableDefinition = new TableDefinition($this->getNativeDatatypeClass());
         $tableDefinition->setName($tableName);
         $tableDefinition->setPrimaryKeysNames($primaryKeys);
         foreach ($columnMetadata as $columnName => $metadata) {
             $tableDefinition->addColumn($columnName, $metadata);
         }
         return $tableDefinition;
     }
+
+    private function getNativeDatatypeClass(): ?string
+    {
+        $dataTupeSource = $this->getDatatypeSourceFromMetadata($this->tableMetadata);
```

Comment:
![](https://th.bing.com/th/id/R.e0d16918f6cd3464cad4d11c11159c93?rik=Wn62NH%2fNHEYctg&pid=ImgRaw&r=0) is this the right tupée?